python_modules/dagster/dagster/_core/definitions/source_asset.py

```diff
@@ -177,41 +178,44 @@ def op(self) -> OpDefinition:
     def is_observable(self) -> bool:
         return self.node_def is not None
 
+    def _get_op_def_compute_fn(self, observe_fn: SourceAssetObserveFunction):
+        from dagster._core.definitions.decorators.solid_decorator import DecoratedOpFunction
+
+        observe_fn_has_context = is_context_provided(observe_fn)
+
+        def fn(context: OpExecutionContext):
+            # This version of is_context_provided does not check if the parameter name is context
```

Comment:
This comment no longer needed with https://github.com/dagster-io/dagster/pull/11982

If that lands then you could add the check if you want by passing `valid_names` kwarg.